### PR TITLE
lock.acquire() must be in the finally close

### PR DESCRIPTION
**Pull request by pypingou**
Otherwise the finally part is not being run I believe

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -96,8 +96,8 @@ Examples
 
     # If you can not use the *with* statement, use a try-finally construct
     # instead:
-    lock.acquire()
     try:
+        lock.acquire()
         pass
```
Comment (pombredanne) [inline docs/source/index.rst]: Would this `pass` still be needed then?

Comment (pypingou) [reply docs/source/index.rst]: Good point, it is likely not :)

**Comment by AdorablePotato** [comment]
Hi, thank you. I think there is some ambiguousness in the examples.

The original example
```
lock.acquire()
try:
   pass
finally:
   lock.release()
```
was equivalent to:
```
with lock.acquire(...):
    pass
```

I'm not sure, but I think what you want with the changed version is something like
```
try:
   lock.acquire()
except TimeoutError:
   # Handle lock timeout.
   pass
else:
  try:
      # Code, that relies on the lock here.
      pass
  finally:
      lock.release()
```

I will change the order of the examples to make things clearer.

**Comment by pypingou** [comment]
Hm ok, I think I see the idea, I would think it might be worth adding an ``# code that relies on the lock here and may throw exceptions`` above the ``pass`` command, as you did in the example you gave here :)
Just for clarity :)

**Pull request closed without merging**